[FEATURE] Allow loading mappings from paths

FoxxMD opened the pull request:
Fluent will check the "paths" node in the doctrine config for an array of directories which it should load mapping files from
### Changes proposed in this pull request:
- Fluent will check the "paths" node in the doctrine config for an array of directories which it should load mapping files from and then load them

FoxxMD committed:
Allow loading mappings from paths

Fluent will check the "paths" node in the doctrine config for an array of directories which it should load mapping files from

guiwoda commented on src/FluentDriver.php:
```diff
@@ -117,6 +127,103 @@ public function getMappers()
+        $getting_namespace = $getting_class = false;
+
+        //Go through each token and evaluate it as necessary
+        foreach (token_get_all($contents) as $token) {
```
@FoxxMD this is why I didn't add the file parser right away. Tokenizing a file's contents seems overkill, when the alternative is to list the FQCNs in the config and be done with it.
I'm not comfortable with the current solution, I don't know what @patrickbrouwers thinks about it.

FoxxMD replied:
The alternative is adding 80+ lines to my doctrine config, for each manager. It gets unwieldy.  Plus with paths I don't have to update my doctrine config each time I add a new mapping, just add the mapping and be done with it.

FoxxMD replied:
Let me see if I can get doctrine's way to work...

FoxxMD committed:
Use require and reflection to determine classes from paths node

Instead of tokenizing just require all files in paths and then use reflection and declared classes to find mapping classes

FoxxMD committed:
Default driver constructor params to null

So all tests pass

FoxxMD commented on src/FluentDriver.php:
```diff
@@ -117,6 +127,45 @@ public function getMappers()
+                }
+
+                $sourceFile = realpath($file->getPathName());
+                require_once $sourceFile;
```
@guiwoda  This is how [doctrine](http://www.doctrine-project.org/api/common/2.4/source-class-Doctrine.Common.Persistence.Mapping.Driver.StaticPHPDriver.html#82-133) finds classes for the StaticPHPDriver

patrickbrouwers commented:
Keeping this open as reminder, but I'd prefer a different solution.

patrickbrouwers commented:
Replaced by #48
